src/datasets/demo_only.py

```diff
@@ -168,42 +169,60 @@ def _generate_demonstrations(
         if idx >= CFG.max_initial_demos:
             break
         try:
-            if CFG.demonstrator == "oracle":
-                timeout = CFG.offline_data_planning_timeout
-                if timeout == -1:
-                    timeout = CFG.timeout
-                oracle_approach.solve(task, timeout=timeout)
-                # Since we're running the oracle approach, we know that
-                # the policy is actually a plan under the hood, and we
-                # can retrieve it with get_last_plan(). We do this
-                # because we want to run the full plan.
-                last_plan = oracle_approach.get_last_plan()
-                policy = utils.option_plan_to_policy(last_plan)
-                # We will stop run_policy() when OptionExecutionFailure()
-                # is hit, which should only happen when the goal has been
-                # reached, as verified by the assertion later.
-                termination_function = lambda s: False
-            else:  # pragma: no cover
-                policy = functools.partial(_human_demonstrator_policy, env,
-                                           idx, num_tasks, task,
-                                           event_to_action)
-                termination_function = task.goal_holds
-            if CFG.make_demo_videos:
-                monitor = utils.VideoMonitor(env.render)
-            else:
-                monitor = None
-            traj, _ = utils.run_policy(
-                policy,
-                env,
-                "train",
-                idx,
-                termination_function=termination_function,
-                max_num_steps=CFG.horizon,
-                exceptions_to_break_on={
-                    utils.OptionExecutionFailure,
-                    utils.HumanDemonstrationFailure,
-                },
-                monitor=monitor)
+            # Will run until we find a plan that successfuly generates a
+            # low-level trajectory that achieves our goal
+            while True:
```

Comment:
this is a potentially large change and it makes me a bit nervous. if we ever have some train tasks that are actually unsolvable (and who knows, maybe we already do) then this will infinitely hang during data collection. why do we need to make this change? I would rather fix up whatever we need to fix within behavior, if oracle + behavior is not robust enough with just a single attempt.

Reply:
ah I see that this only matters for behavior because it will break after one attempt in the usual case. so i'm less nervous about impacting existing code, but I still wonder if we can find another way to address this

Reply:
We'll remove it for this PR.